hibernate-core/src/main/java/org/hibernate/boot/SessionFactoryBuilder.java

```diff
@@ -705,6 +706,17 @@ default SessionFactoryBuilder applyConnectionProviderDisablesAutoCommit(boolean
 	SessionFactoryBuilder enableReleaseResourcesOnCloseEnabled(boolean enable);
 
 	/**
+	 * Specify the CustomLoaderFactory to use.
+	 *
+	 * @param customLoaderFactory The CustomLoaderFactory to use.
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	default SessionFactoryBuilder applyCustomLoaderFactory(CustomLoaderFactory customLoaderFactory) {
```

Comment:
I don't think I am a fan of these kind of method defaults.  Here the default really does not do what it is documented to do - meaning that an implementation inheriting this method (thats what method defaults are for after all) will not work correctly.

Reply:
Additionally, considering this stuff all goes away in 6 I am not really a fan of making this "user facing".  *If* we do end up exposing it to users, these methods need to be deprecated.